### PR TITLE
fix: Correct the code to move to login page

### LINKED ISSUE #2047: Clicking on "Please login to post feedback" opens ChatActivity instead of login activity
**Describe the bug**
Clicking on "Please login to post feedback" in the SkillDetailsFragment opens ChatActivity instead of login activity.

**Expected Behaviour**
It should open LoginActivity instead of ChatActivity.

**Steps to reproduce it**
Modify the intent. Use IsettingsPresenter and its loginLogout function to resolve it.

**Screenshots of the issue**
feedback

**Smartphone Info:**

|               | |
|---------------|-|
|Device         | Redmi Note 5 Pro |
|Android Version|< Oreo 8.1    |

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2047 2047

Changes:
Added the correct code in killDetailsFragment.kt

Screenshots for the change: 
workingfeedback

### CODE REVIEW

**Comment by atm1504** [comment]
@iamareebjamal please check wether the method is correct or not. If not shall I use the latest method that I have used in #2051

**Comment by iamareebjamal** [comment]
Don't show login if already logged in

**Comment by atm1504** [comment]
> Don't show login if already logged in

Sir, its not showing login if the user has logged in before.

**Comment by atm1504** [comment]
@iamareebjamal @batbrain7 please review it

**Comment by atm1504** [comment]
@batbrain7 please review it

**Comment by atm1504** [comment]
@batbrain7 @iamareebjamal @arundhati24 please review it

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 1]
```diff
@@ -499,7 +504,9 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
         } else {
             tvAnonymousPostFeedback.visibility = View.VISIBLE
             tvAnonymousPostFeedback.setOnClickListener {
+                settingsPresenter.loginLogout()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: As I said earlier, this is wrong

**Comment by atm1504** [comment]
@iamareebjamal please review it now. I have corrected the code

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt [hunk 2]
```diff
@@ -33,11 +36,21 @@ class SkillDetailsPresenter(skillDetailsFragment: SkillDetailsFragment) : ISkill
...
     private var skillDetailsView: ISkillDetailsView? = null
     private val utilModel: UtilModel = UtilModel(skillDetailsFragment.requireContext())
+    private var databaseRepository: IDatabaseRepository = DatabaseRepository()
+    private var settingView: ISettingsView? = null
 
     override fun onAttach(skillDetailsView: ISkillDetailsView) {
         this.skillDetailsView = skillDetailsView
     }
 
+    override fun loginLogout() {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: Duplicate. Extract the function in common module and use in both presenters

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: @iamareebjamal can you please explain how to do this?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: B has a method

You want to call it in D

Create a component A which has that method, 

import A in B and in D

**Comment by atm1504** [comment]
Fixes: #2156 #2047

Created loginLogoutModule presenter and used this in every where login logout is necessary. This has reduced the duplicity of the function definition. Moreover this presenter class only deals with login logout feature. So it can be used anywhere, where login logout is required.

**Comment by atm1504** [comment]
@iamareebjamal @batbrain7 @arundhati24 please review it.

File: app/src/main/java/org/fossasia/susi/ai/login/LoginLogoutModulePresenter.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package org.fossasia.susi.ai.login
+
+import android.app.Activity
+import org.fossasia.susi.ai.data.UtilModel
+import org.fossasia.susi.ai.data.db.DatabaseRepository
+import org.fossasia.susi.ai.data.db.contract.IDatabaseRepository
+import org.fossasia.susi.ai.login.contract.ILoginLogoutModulePresenter
+import org.fossasia.susi.ai.skills.settings.contract.ISettingsView
+
+class LoginLogoutModulePresenter(activeActivity: Activity) : ILoginLogoutModulePresenter {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginLogoutModulePresenter.kt]: Why do you need Activity, why not context?

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginLogoutModulePresenter.kt]: Sir,I have changed it. Please review now.

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt [hunk 4]
```diff
@@ -81,7 +82,7 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
         if (thisActivity is SkillsActivity) thisActivity.title = getString(R.string.action_settings)
         settingsPresenter = SettingsPresenter(activity as SkillsActivity)
         settingsPresenter.onAttach(this)
-        loginLogoutModulePresenter = LoginLogoutModulePresenter(activity as SkillsActivity)
+        loginLogoutModulePresenter = LoginLogoutModulePresenter(context as Context)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt]: No need to cast, use requireContext()

**Comment by atm1504** [comment]
@iamareebjamal @batbrain7 @arundhati24 please review it now.

File: app/src/main/java/org/fossasia/susi/ai/login/LoginLogoutModulePresenter.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.fossasia.susi.ai.data.db.DatabaseRepository
+import org.fossasia.susi.ai.data.db.contract.IDatabaseRepository
+import org.fossasia.susi.ai.login.contract.ILoginLogoutModulePresenter
+import org.fossasia.susi.ai.skills.settings.contract.ISettingsView
+
+class LoginLogoutModulePresenter(context: Context) : ILoginLogoutModulePresenter {
+
+    private var utilModel: UtilModel = UtilModel(context)
+    private var databaseRepository: IDatabaseRepository = DatabaseRepository()
+    private var settingView: ISettingsView? = null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginLogoutModulePresenter.kt]: Why var?

File: app/src/main/java/org/fossasia/susi/ai/login/contract/ILoginLogoutModulePresenter.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package org.fossasia.susi.ai.login.contract
+
+interface ILoginLogoutModulePresenter {
+    fun loginLogout()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/contract/ILoginLogoutModulePresenter.kt]: logout

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/login/contract/ILoginLogoutModulePresenter.kt]: Sir, I didn't understand what you wanted to say by logout. Please explain.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/login/contract/ILoginLogoutModulePresenter.kt]: Change method name

**Comment by atm1504** [comment]
@iamareebjamal please review now. I have made the suggested changes.

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt [hunk 7]
```diff
@@ -137,14 +137,6 @@ class SignUpPresenter(signUpActivity: SignUpActivity) : ISignUpPresenter, ISignU
         signUpView = null
     }
 
-    override fun loginLogout() {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: This will remain here but delegated to the new presenter

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: @batbrain7 @arundhati24 @iamareebjamal what is the need of keeping this here when we have a separate presenter file that handles all the login operations? Isn't it just writing just extra lines of codes?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: Read about delegation

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: And no, no lines will be repeated

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpPresenter.kt]: I will surely look into it sir

**Comment by atm1504** [comment]
@iamareebjamal please check now. Please suggest if this is the required change that you wanted.

**Comment by atm1504** [comment]
@iamareebjamal please review it

**Comment by iamareebjamal** [comment]
Build is failing

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 8]
```diff
@@ -33,6 +35,7 @@ import org.fossasia.susi.ai.skills.SkillsActivity
 class SignUpActivity : AppCompatActivity(), ISignUpView {
 
     private lateinit var signUpPresenter: ISignUpPresenter
+    private lateinit var loginLogoutModulePresenter: ILoginLogoutModulePresenter
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: Nothing in this Activity should change as mentioned before

**Comment by mariobehling** [comment]
Merging this as a move forward.

**Pull request merged**